libr/core/cmd_search.c

```diff
@@ -2296,31 +2337,47 @@ static int cmd_search(void *data, const char *input) {
 	param.rsa_search = false;
 	param.use_mread = false;
 	param.do_bckwrd_srch = false;
+	param.boundaries = NULL;
 
 	c = 0;
 	json = false;
 	first_hit = true;
 	// core->search->n_kws = 0;
 	maplist = false;
-	__from = r_config_get_i (core->config, "search.from");
-	__to = r_config_get_i (core->config, "search.to");
+	config_from = r_config_get_i (core->config, "search.from");
+	config_to = r_config_get_i (core->config, "search.to");
 
 	searchshow = r_config_get_i (core->config, "search.show");
 	param.mode = r_config_get (core->config, "search.in");
 	param.boundaries = r_core_get_boundaries (core, param.mode,
 		&param.from, &param.to);
 	param.use_mread = (!strcmp (param.mode, "maps"))? 1: 0;
 
-	if (__from != UT64_MAX) {
-		param.from = __from;
+	if (!strcmp (param.mode, "io.maps") && param.boundaries != NULL &&
```

Comment:
It's not boolean, therefore is not treated as such. I actually value readability.